### PR TITLE
scripts: serie_update:  minor fixes to be able to execute the script on ubuntu+zsh

### PR DESCRIPTION
**Pull request by str4t0m**
With my setup(ubuntu+zsh) the script crashes, when subprocess.check_call()
parameters are passed as list and shell=True is set,
or if subprocess.check_call() parameters are passed as string
and shell=True is not set. These changes fix that.

I did not yet have a chance to test if the changes break anything on Windows.

### CODE REVIEW

**Review by ABOSTM — APPROVED** [review]
I tested this PR on windows, and it is working fine.
LGTM

**Comment by str4t0m** [comment]
Thanks for testing this on windows.

File: scripts/serie_update.py [hunk 1]
```diff
@@ -30,6 +30,7 @@ def os_cmd(cmd, cwd=None, shell=False):
     Args:
         cmd: string command to execute.
         cwd: directory where to run command
+        shell: boolean to enable command interpretation by the shell
```
Comment (erwango) [inline scripts/serie_update.py:33]: Any impact on ./update_stm32_package.py ?
Unless this is totally transparent to users in any case, please update scripts/README.rst ?

**Review by erwango — CHANGES_REQUESTED** [review]

File: scripts/serie_update.py (hunk 1 above)
Comment (str4t0m) [reply scripts/serie_update.py:33]: The function was just missing the arguments doc string, so I decided to include it in this commit.
Functionally nothing changed, can I leave it as it is?

**Review by erwango — APPROVED** [review]

File: scripts/serie_update.py [hunk 2]
```diff
@@ -264,12 +265,9 @@ def extract_source(self):
...
         os_cmd(
-            (
-                "cp",
-                "-r",
-                str(stm32cube_drivers_src_path) + "/*.*",
-                str(temp_drivers_src_path),
-            )
+            "cp -r " + str(stm32cube_drivers_src_path) + "/*.* "
+            + str(temp_drivers_src_path),
+            shell=True,
```
Comment (gmarull) [inline scripts/serie_update.py:270]: I understand that this is in fact the only change required, others are cosmetic and not strictly required (should go into a separate commit)

File: scripts/serie_update.py [hunk 3]
```diff
@@ -355,15 +345,8 @@ def build_patch_from_current_zephyr_version(self):
...
-                (
-                    "git",
-                    "diff",
-                    "HEAD@{1}",
-                    "--",
-                    str(hal_conf),
-                    ">>",
-                    str(hal_conf_patch),
-                ),
+                "git diff HEAD@{1} -- " + str(hal_conf)
```
Comment (gmarull) [inline scripts/serie_update.py:348]: in general it is better to use multi-line strings if they do not fit in one line, so I prefer the previous code in this case, others LGTM

**Review by gmarull — CHANGES_REQUESTED** [review]
Main request: split into 2 commits (actual fix and cosmetic changes)

File: scripts/serie_update.py (hunk 3 above)
Comment (str4t0m) [reply scripts/serie_update.py:348]: Unfortunately, except the comment in line 33, this is the minimal set of changes to get it running.
The arguments are treated differently on unix systems than on windows.

[Popen Constructor](https://docs.python.org/2/library/subprocess.html#popen-constructor):
_"On Unix with shell=True,   . If args is a sequence, the first item specifies the command string, and any additional items will be treated as additional arguments to the shell itself"_

In this case "git" is executed without arguments, and the rest of the sequence are arguments passed to the shell.
I am afraid that there is no way on unix systems to avoid passing the whole command as a string for commands that need "shell=True". But if you know any better solution that could work I will try it.

Comment (erwango) [reply scripts/serie_update.py:348]: @gmarull I confirm this is working. Is that blocking ?

**Review by gmarull — COMMENTED** [review]
I see the problem, but instead of merging long strings, better do:
```
a = (
   "b" +
   "c" +
   ...
   "z"
)
```

when the command is too long. That is, replace `,` with `+`. Anyway I think that current script deserves a re-write, so I won't block.

**Comment by str4t0m** [comment]
I have reformatted it now as suggested.

**Comment by erwango** [comment]
@galak Can we merge this one ?
